01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+    y = "Density"
+  )
+
+
+house_value_place_subgroup |>
+  ggplot() +
+  geom_density(aes(x = share_house_value),
+    color = "blue", fill = alpha("blue", 0.3)
+  ) +
+  facet_wrap(race_ethnicity ~ subgroup, ncol = 2) +
+  theme_minimal() +
+  labs(
+    title = "Share House Value by Race/Ethnicity and Age",
+    subtitle = "Place, Age over 45 Subgroup",
+    x = "Share House Value",
+    y = "Density"
+  )
+```
+
+## Create Final Output Data: Merge Household & Housing Value Share
+
+```{r}
+#| label: compute-ratio-house-value-households
+
+construct_ratio <- function(share_house_value, share_households) {
+  paste0(
+    format(round(100 * share_house_value, 1), nsmall = 1),
+    "%",
+    ":",
+    format(round(100 * share_households, 1), nsmall = 1),
+    "%"
+  ) |>
+    str_remove_all(pattern = " ")
+}
+
+ratio_house_value_households <- function(households_data, house_value_data, ...) {
+  geo_var <- if_else("county" %in% colnames(households_data), "county", "place")
+  households_data |>
+    left_join(house_value_data,
+      by = c("year", "state", geo_var, "race_ethnicity", ...)
+    ) |>
+    mutate(
+      ratio_house_value_households = construct_ratio(share_house_value, share_households),
+      .by = c(race_ethnicity, ...)
+    ) |>
+    arrange(year, state, {{ geo_var }}, ...)
+}
+
+
+reshape_to_wide <- function(data, ...) {
+  data |>
+    pivot_wider(
+      id_cols = c(year, state, ...),
+      names_from = race_ethnicity,
+      values_from = c(ratio_house_value_households, ratio_house_value_households_quality),
+      names_glue = "{race_ethnicity}_{.value}"
+    ) %>%
+    rename_with(~ str_remove(.x, "ratio_") %>% paste0("ratio_", .), contains("ratio")) %>%
+    # order variables according to request
+    select(
+      year, state, ...,
+      ratio_black_nh_house_value_households, ratio_black_nh_house_value_households_quality,
+      ratio_hispanic_house_value_households, ratio_hispanic_house_value_households_quality,
+      ratio_other_nh_house_value_households, ratio_other_nh_house_value_households_quality,
+      ratio_white_nh_house_value_households, ratio_white_nh_house_value_households_quality
+    ) %>%
+    arrange(year, state, ...)
+}
+```
+
+```{r}
+#| label: calculate-metric
+
+ratio_house_value_households_county <- ratio_house_value_households(
+  share_households_county,
+  house_value_county
+) |>
+  reshape_to_wide(county)
+
+
+ratio_house_value_households_county_subgroup <- ratio_house_value_households(
+  share_households_county_subgroup,
+  house_value_county_subgroup,
+  "subgroup"
+) |>
+  reshape_to_wide(county, subgroup)
+
+ratio_house_value_households_place <-
+  ratio_house_value_households(
+    share_households_place,
+    house_value_place
+  ) |>
+  reshape_to_wide(place)
+
+ratio_house_value_households_place_subgroup <-
+  ratio_house_value_households(
+    share_households_place_subgroup,
+    house_value_place_subgroup,
+    "subgroup"
+  ) |>
+  reshape_to_wide(place, subgroup)
+```
```

Comment:
For these merges, there was no information on whether you expected perfect matches (i.e., all observations to merge) for these two datasets. I would add unit tests to confirm unmatched data from x is what you expected to be unmatched (as a failsafe) and also add text above the matches to describe why the matches are working as expected.